model/task/task.go

```diff
@@ -3669,6 +3669,58 @@ func GetGroupedTaskStatsByVersion(versionID string, opts GetTasksByVersionOption
 
 }
 
+// GetMatchingBaseTasks returns the corresponding base tasks for tasks on a version.
+// If a task ran on the base version but was not scheduled on the version, then it won't be returned.
+func GetMatchingBaseTasks(versionID string, baseVersionID string) ([]Task, error) {
```

Comment:
We technically don't need to return all of the tasks here we only need their statuses. It would be prefferable if we filtered out the statuses in the db and just return the already unique statuses instead of a full task doc.